templates/security.conf.erb

```diff
@@ -9,7 +9,9 @@
 # Set to one of:  Full | OS | Minimal | Minor | Major | Prod
 # where Full conveys the most information, and Prod the least.
 #
+<% unless @server_tokens.nil? %>
```

Comment:
I don't think this is needed at all. When this is set to `nil` this block does not work as expected (same in all places).

Reply:
So, remove the `nil` check from each of them?

Reply:
The reason that I added those is because the resource can be called directly, with overridden options, and if one of those values is not set than the resulting config would be invalid.

Reply:
Can you please give a code example that would make this happen?

Reply:
Sure. The default `options` value for the `conf` resource includes a hash of:

```ruby
{
  server_tokens: 'Prod',
  server_signature: 'On',
  trace_enable: 'Off',
}
```

If the options are overridden with this call:

```ruby
apache2_conf 'security' do
  options(
    server_tokens: 'Full',
    trace_enable: 'Off'
  )
end
```

the `server_signature` value is `nil` and the resulting _security.conf_ file will have the a line like

```
ServerSignature 
```

rather than

```
ServerSignature On
```

Reply:
If your preference is that the config _should_ fail if those values are omitted I'm happy to remove the checks.

Reply:
Ah yes, I see what you mean. I was only trying from the `apache2_install` resource, not from the `apache2_conf` resource directly. Thanks for pointing that out.